### PR TITLE
14350

### LINKED ISSUE #14350: DOC/ERR: update error message / doc-string for to_datetime with non-convertible object and unit kw
#### A small, complete example of the issue

```
import pandas as pd
pd.to_datetime(datetime.datetime(2016,1,1), unit='s')
Traceback (most recent call last):
  File "/home/julienv/.pycharm_helpers/pydev/pydevd_exec2.py", line 3, in Exec
    exec(exp, global_vars, local_vars)
  File "<input>", line 1, in <module>
  File "/usr/local/lib/python3.4/dist-packages/pandas/util/decorators.py", line 91, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.4/dist-packages/pandas/tseries/tools.py", line 424, in to_datetime
    return _convert_listlike(np.array([arg]), box, format)[0]
  File "/usr/local/lib/python3.4/dist-packages/pandas/tseries/tools.py", line 330, in _convert_listlike
    errors=errors)
  File "pandas/tslib.pyx", line 2144, in pandas.tslib.array_with_unit_to_datetime (pandas/tslib.c:39248)
  File "pandas/tslib.pyx", line 2255, in pandas.tslib.array_with_unit_to_datetime (pandas/tslib.c:38492)
ValueError: non convertible value 2016-01-01 00:00:00with the unit 's'
```
#### Expected Output

Timestamp('2016-01-01 00:00:00')
#### Output of `pd.show_versions()`

<details>

> > > pd.show_versions()
## INSTALLED VERSIONS

commit: None
python: 3.4.3.final.0
python-bits: 64
OS: Linux
OS-release: 3.13.0-96-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: fr_FR.UTF-8
LOCALE: fr_FR.UTF-8

pandas: 0.19.0
nose: None
pip: 1.5.4
setuptools: 3.3
Cython: 0.20.1post0
numpy: 1.11.2
scipy: 0.18.1
statsmodels: None
xarray: None
IPython: None
sphinx: 1.2.2
patsy: None
dateutil: 2.5.3
pytz: 2016.7
blosc: None
bottleneck: None
tables: None
numexpr: None
matplotlib: 1.5.1
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: 0.999
httplib2: None
apiclient: None
sqlalchemy: None
pymysql: None
psycopg2: 2.5.3 (dt dec pq3 ext)
jinja2: 2.7.3
boto: None
pandas_datareader: None
</details>

### PR DESCRIPTION
**Pull request by mpenkov**
- [x] closes #14350
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
thanks!

**Comment by jorisvandenbossche** [comment]
@mpenkov in general, can you try to give PRs a more informative title ?

File: doc/source/whatsnew/v0.22.0.txt [hunk 1]
```diff
@@ -26,6 +26,7 @@ Other Enhancements
 - :func:`pandas.tseries.frequencies.to_offset` now accepts leading '+' signs e.g. '+1h'. (:issue:`18171`)
 - :class:`pandas.io.formats.style.Styler` now has method ``hide_index()`` to determine whether the index will be rendered in ouptut (:issue:`14194`)
 - :class:`pandas.io.formats.style.Styler` now has method ``hide_columns()`` to determine whether columns will be hidden in output (:issue:`14194`)
+- Improved wording of ValueError raised in :func:`Timestamp.tz_localize` function
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.22.0.txt]: Why is this mentioning `tz_localize`? Wasn't this about `to_datetime` ?

Comment (mpenkov) [reply doc/source/whatsnew/v0.22.0.txt]: You're right, I specified the wrong function name in whatsnew. I pushed an update just now.

Comment (jreback) [reply doc/source/whatsnew/v0.22.0.txt]: hah i merged already
i’ll fix

**Pull request merged**